realm-jni/src/io_realm_internal_SharedGroup.cpp

```diff
@@ -295,3 +304,20 @@ JNIEXPORT jboolean JNICALL Java_io_realm_internal_SharedGroup_nativeCompact(
     CATCH_STD()
     return false;
 }
+
+JNIEXPORT jlongArray JNICALL Java_io_realm_internal_SharedGroup_nativeGetVersionID
+        (JNIEnv *env, jobject, jlong native_ptr)
+{
+    TR_ENTER()
+
+    SharedGroup::VersionID versionID = SG(native_ptr)->get_version_of_current_transaction();
+
+    jlong versionArray [2];
+    versionArray[0] = static_cast<jlong>(versionID.version);
+    versionArray[1] = static_cast<jlong>(versionID.index);
+
+    jlongArray versionData = env->NewLongArray(2);
```

Comment:
`NewLongArray()` might run out of memory so we need a `try`/`catch` block to guard it.